persistence/trees/trees.go

```diff
@@ -251,6 +272,66 @@ func (t *treeStore) getStateHash() string {
 	return hexHash
 }
 
+////////////////////////////////
+// AtomicStore Implementation //
+////////////////////////////////
+
+// Savepoint generates a new savepoint for the tree store and saves it internally.
+func (t *treeStore) Savepoint() error {
+	w, err := t.save()
+	if err != nil {
+		return err
+	}
+	t.Prev = w
+	return nil
+}
+
+// Rollback intentionally can't return an error because at this point we're out of tricks
+// to recover from problems.
+func (t *treeStore) Rollback() {
+	if t.Prev != nil {
+		t.merkleTrees = t.Prev.MerkleTrees
```

Comment:
Do we need to do any sort of release or is it all magic?

Reply:
Not as far as I know - @h5law can speak more to the details of the `smt` side, and correct me if I'm wrong about anything, but as far as I understand, the pointer will get dropped if it's not used, and in the case of a rollback being triggered, the reassignment kicks in which doesn't see the uncommitted tree changes because they haven't been written to disk, and it uses the same underlying connection from the `nodeStore`. Pretty slick, all things considered.

Reply:
Can you just leave a comment next to the code explaining this?